File: src/main/java/model/openloopfueling/util/LogUtil.java
package model.openloopfueling.util;

import java.util.List;

public class LogUtil {
    static boolean isValidLogLength(int start, int minPoints, double minThrottleAngle, List<Double> throttleAngle) {
        int minValidIndex = start + minPoints;

        new Exception().printStackTrace();

        if (minValidIndex < throttleAngle.size()) {
            for (int i = start; i < minValidIndex; i++) {
                if (throttleAngle.get(i) <= minThrottleAngle) {
                    return false;
                }
            }

            return true;
        }

        return false;
    }

    static int findEndOfLog(int start, double minThrottleAngle, List<Double> thottleAngle) {

        for (int i = start; i < thottleAngle.size(); i++) {
            if (thottleAngle.get(i) < minThrottleAngle) {
                return i;
            }
        }

        return thottleAngle.size() - 1;
    }
}

File: src/main/kotlin/domain/model/openloopfueling/util/LogUtil.kt
package domain.model.openloopfueling.util

object LogUtil {
    fun isValidLogLength(start: Int, minPoints: Int, minThrottleAngle: Double, throttleAngle: List<Double>): Boolean {
        val minValidIndex = start + minPoints
        if (minValidIndex < throttleAngle.size) {
            for (i in start until minValidIndex) {
                if (throttleAngle[i] <= minThrottleAngle) return false
            }
            return true
        }
        return false
    }

    fun findEndOfLog(start: Int, minThrottleAngle: Double, throttleAngle: List<Double>): Int {
        for (i in start until throttleAngle.size) {
            if (throttleAngle[i] < minThrottleAngle) return i
        }
        return throttleAngle.size - 1
    }
}
